docs/extensions.adoc

```diff
@@ -723,6 +723,95 @@ runner {
 }
 ----
 
+[[snapshot-testing]]
+=== Snapshot testing
+
+Since version 2.4, the `@Snapshot` extension has been added for snapshot testing.
+This type of testing compares the current output of your code with a saved version (a snapshot) to check for any changes.
+It's especially useful for checking if your code still produces the right text outputs after changes.
+The `@Snapshot` extension also makes it easy to create and update these snapshots.
+
+You use the `@Snapshot` extension along with the `Snapshotter` class.
+The `Snapshotter` class provides the entrypoint for the tests.
+You can use `@Snapshot` in two ways: you can add it to a field in your code, or you can use it as a parameter in a method.
+The examples below will show you how to do both.
+
+.Snapshot Extension as field
+[source,groovy]
+----
+include::{sourcedir}/extension/SnapshotDocSpec.groovy[tag=field]
+----
+
+.Snapshot Extension as parameter
+[source,groovy]
+----
+include::{sourcedir}/extension/SnapshotDocSpec.groovy[tag=parameter]
+----
+
+You can store and assert over multiple snapshots by providing an optional snapshotId as identifier.
+
+.Multiple snapshots
+[source,groovy]
+----
+include::{sourcedir}/extension/SnapshotDocSpec.groovy[tag=multi-snapshot]
+----
+
+By default, snapshots are stored and compared as plain text.
+However, you can plug your own matching logic.
+For example, you could compare two json structures with a third-party library.
+
+.Custom matching
+[source,groovy]
+----
+include::{sourcedir}/extension/SnapshotDocSpec.groovy[tag=custom-matching]
+----
+
+You can also extend from the `Snapshotter` class to provide your own convenience methods.
+Check `org.spockframework.specs.extension.SpockSnapshotter` in the spock codebase for an example.
+
+==== Configuration
+The snapshots are stored in the `rootPath` directory, which can be configured either in the  <<spock-configuration-file,Spock Configuration File>>, or via the `spock.snapshots.rootPath` system property.
+The `rootPath` directory is required and the `@Snapshot` extension will throw an exception if it is not configured when the extension is used.
+
+Snapshots can be updated when setting the `spock.snapshots.updateSnapshots` system property to `true`, or via the config file.
+
+You can configure the extension to store the actual value in case of a mismatch by setting the `spock.snapshots.writeActual`  system property to `true`, or via the config file.
```

Comment:
```suggestion
You can configure the extension to store the actual value in case of a mismatch by setting the `spock.snapshots.writeActual` system property to `true`, or via the config file.
```